pandas/core/indexes/datetimelike.py

```diff
@@ -532,6 +533,64 @@ def isin(self, values):
 
         return algorithms.isin(self.asi8, values.asi8)
 
+    def intersection(self, other, sort=False):
+        self._validate_sort_keyword(sort)
+        self._assert_can_do_setop(other)
+
+        if self.equals(other):
+            return self._get_reconciled_name_object(other)
+
+        if not isinstance(other, type(self)):
+            try:
+                other = self(other)
+            except (TypeError, ValueError):
+                pass
```

Comment:
Can you add a comment explaining the `try-except` here.

Reply:
This came from a previous piece of code. I added comments trying to explain it. Happy for feedback if this needs more changing.

Reply:
hmm, does anything break if you remove the entire try/except here, e.g. we simply call ``Index.intersection`` directrly